genie-proto/src/main/proto/genie.proto

```diff
@@ -133,3 +133,56 @@ message JobSpecificationResponse {
         JobSpecificationServiceError error = 2;
     }
 }
+
+//
+// Agent job execution events service
+//
+service JobExecutionEventService {
+    rpc publishEvent (JobExecutionEvent) returns (JobExecutionEventAcknowledgment);
+}
+
+//
+// Outer envelope for job execution events and server acknowledgement
+//
+message JobExecutionEvent {
+    string agent_id = 1;
+    string job_id = 2;
+    google.protobuf.Timestamp timestamp = 3;
+    oneof event_type {
+        StateChangeEvent state_change_event = 10;
+        JobCompletionEvent job_completion_event = 11;
+        StateActionCompletionEvent state_action_completion_event = 12;
+    }
+}
+
+message JobExecutionEventAcknowledgment {
+}
+
+//
+// Job execution event types
+//
+message StateChangeEvent {
+    string from_state = 1;
```

Comment:
I'm not sure yet, i think we might revisit this once we decide how to do server-side tracking.

If the server is not aware of the agent state machine, the two are more decoupled and may be easier to upgrade agents without needing to sync with the server.
On the other side, the agent state machine is probably not changing, and it would be nice to leave the agent smart and lightweight, just post simple events and let the server interpret them (e.g. map them to macro states exposed to users)